File: server/src/test/java/com/adobe/testing/s3mock/store/StoreTestBase.java
/*
 *  Copyright 2017-2022 Adobe.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.adobe.testing.s3mock.store;

import static java.util.Collections.emptyMap;

import com.adobe.testing.s3mock.dto.Owner;
import java.io.File;
import java.nio.file.Paths;
import java.util.Map;
import java.util.UUID;
import org.apache.http.entity.ContentType;
import org.springframework.beans.factory.annotation.Autowired;

abstract class StoreTestBase {
  static final String TEST_BUCKET_NAME = "test-bucket";
  static final String TEST_FILE_PATH = "src/test/resources/sampleFile.txt";
  static final String NO_ENC = null;
  static final String NO_ENC_KEY = null;
  static final Map<String, String> NO_USER_METADATA = emptyMap();
  static final String TEST_ENC_TYPE = "aws:kms";
  static final String TEST_ENC_KEY = "aws:kms" + UUID.randomUUID();
  static final String TEXT_PLAIN = ContentType.TEXT_PLAIN.toString();
  static final String ENCODING_GZIP = "gzip";
  static final String NO_PREFIX = null;
  static final String DEFAULT_CONTENT_TYPE =
      ContentType.APPLICATION_OCTET_STREAM.toString();
  static final Owner TEST_OWNER = new Owner("123", "s3-mock-file-store");

  @Autowired
  private File rootFolder;

  BucketMetadata metadataFrom(String bucketName) {
    BucketMetadata metadata = new BucketMetadata();
    metadata.setName(bucketName);
    metadata.setPath(Paths.get(rootFolder.toString(), bucketName));
    return metadata;
  }

}

File: server/src/test/kotlin/com/adobe/testing/s3mock/store/StoreTestBase.kt
/*
 *  Copyright 2017-2025 Adobe.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.adobe.testing.s3mock.store

import com.adobe.testing.s3mock.dto.ChecksumAlgorithm
import com.adobe.testing.s3mock.dto.ChecksumType
import com.adobe.testing.s3mock.dto.Initiator
import com.adobe.testing.s3mock.dto.ObjectOwnership
import com.adobe.testing.s3mock.dto.Owner
import com.adobe.testing.s3mock.dto.Tag
import com.adobe.testing.s3mock.util.AwsHttpHeaders
import org.apache.http.entity.ContentType
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpHeaders
import org.springframework.http.MediaType
import java.io.File
import java.util.Date
import java.util.UUID

internal abstract class StoreTestBase {
  @Autowired
  private lateinit var rootFolder: File

  protected fun metadataFrom(bucketName: String): BucketMetadata {
    BUCKET_NAMES.add(bucketName)
    return BucketMetadata(
      bucketName,
      Date().toString(),
      null,
      null,
      null,
      ObjectOwnership.BUCKET_OWNER_ENFORCED,
      rootFolder.toPath().resolve(bucketName),
      "us-east-1",
      null,
      null,
    )
  }

  protected fun encryptionHeaders(): Map<String, String> = mapOf(
    AwsHttpHeaders.X_AMZ_SERVER_SIDE_ENCRYPTION to TEST_ENC_TYPE,
    AwsHttpHeaders.X_AMZ_SERVER_SIDE_ENCRYPTION_AWS_KMS_KEY_ID to TEST_ENC_KEY
  )

  protected fun storeHeaders(): Map<String, String> = mapOf(HttpHeaders.CONTENT_ENCODING to ENCODING_GZIP)

  companion object {
    const val TEST_BUCKET_NAME = "test-bucket"
    const val TEST_FILE_PATH = "src/test/resources/sampleFile.txt"

    val NO_USER_METADATA: Map<String, String> = emptyMap()
    val NO_ENCRYPTION_HEADERS: Map<String, String> = emptyMap()
    val NO_TAGS: List<Tag> = emptyList()
    val NO_CHECKSUMTYPE: ChecksumType? = null
    val NO_CHECKSUM: String? = null
    val NO_CHECKSUM_ALGORITHM: ChecksumAlgorithm? = null
    const val TEST_ENC_TYPE = "aws:kms"

    val TEST_ENC_KEY: String = "aws:kms${UUID.randomUUID()}"

    val TEXT_PLAIN: String = ContentType.TEXT_PLAIN.toString()
    const val ENCODING_GZIP = "gzip"
    val NO_PREFIX: String? = null
    const val DEFAULT_CONTENT_TYPE = MediaType.APPLICATION_OCTET_STREAM_VALUE
    val TEST_OWNER: Owner = Owner("123")
    val TEST_INITIATOR: Initiator = Initiator("s3-mock-file-store", "123")

    val BUCKET_NAMES = mutableSetOf<String>()
  }
}
